kotest-assertions/kotest-assertions-shared/src/commonMain/kotlin/io/kotest/assertions/show/Show.kt

```diff
@@ -32,27 +72,13 @@ fun <T : Any> showFor(t: T): Show<T> = platformShow(t) ?: commonShowFor(t)
 expect fun <A : Any> platformShow(a: A): Show<A>?
 
 @Suppress("UNCHECKED_CAST")
-fun <T : Any> commonShowFor(t: T): Show<T> = when (t) {
-   is String -> StringShow as Show<T>
-   is Map<*, *> -> MapShow as Show<T>
-   is BooleanArray -> ArrayShow
-   is IntArray -> ArrayShow
-   is ShortArray -> ArrayShow
-   is FloatArray -> ArrayShow
-   is DoubleArray -> ArrayShow
-   is LongArray -> ArrayShow
-   is ByteArray -> ArrayShow
-   is CharArray -> ArrayShow
-   is Array<*> -> ArrayShow
-   is List<*> -> ListShow<T>() as Show<T>
-   is Iterable<*> -> IterableShow<T>() as Show<T>
-   is Long, t is Boolean, t is Int, t is Double, t is Float, t is Short, t is Byte -> DefaultShow
-   is KClass<*> -> KClassShow as Show<T>
-   else -> when {
-      // this won't work in JS or native, so they'll get the boring old toString version
-      io.kotest.mpp.reflection.isDataClass(t::class) -> dataClassShow<T>()
-      else -> DefaultShow
-   }
+fun <T : Any> commonShowFor(t: T): Show<T> {
+   // lookup a show from the registered typeclasses
+   val show = Shows.all().keys.firstOrNull { it.isInstance(t) }
+   if (show != null) return show as Show<T>
```

Comment:
We need the value of `Show` but here we are casting the `Key`.

Reply:
good spot thanks

Reply:
updated